mlflow/store/artifact/dbfs_artifact_repo.py

```diff
@@ -171,18 +181,25 @@ def dbfs_artifact_repo_factory(artifact_uri):
     :param artifact_uri: DBFS root artifact URI (string).
     :return: Subclass of ArtifactRepository capable of storing artifacts on DBFS.
     """
+    if not is_valid_dbfs_uri(artifact_uri):
+        raise MlflowException("DBFS URI must be of the form dbfs:/<path> or " +
+                              "dbfs://profile@databricks/<path>, but received " + artifact_uri)
+
     cleaned_artifact_uri = artifact_uri.rstrip('/')
-    uri_scheme = get_uri_scheme(artifact_uri)
-    if uri_scheme != 'dbfs':
-        raise MlflowException("DBFS URI must be of the form "
-                              "dbfs:/<path>, but received {uri}".format(uri=artifact_uri))
+    db_profile_uri = get_databricks_profile_uri_from_artifact_uri(cleaned_artifact_uri)
     if is_databricks_acled_artifacts_uri(artifact_uri):
         return DatabricksArtifactRepository(cleaned_artifact_uri)
     elif mlflow.utils.databricks_utils.is_dbfs_fuse_available() \
             and os.environ.get(USE_FUSE_ENV_VAR, "").lower() != "false" \
-            and not artifact_uri.startswith("dbfs:/databricks/mlflow-registry"):
-        # If the DBFS FUSE mount is available, write artifacts directly to /dbfs/... using
-        # local filesystem APIs
-        file_uri = "file:///dbfs/{}".format(strip_prefix(cleaned_artifact_uri, "dbfs:/"))
+            and not artifact_uri.startswith("dbfs:/databricks/mlflow-registry") \
+            and (db_profile_uri is None or db_profile_uri == 'databricks'):
```

Comment:
Just confirming my understanding: this check means that the profile refers to the current workspace right?

Reply:
yeah